### DISCUSSION TITLE
Branch dunfell after flash no micro USB is detected

### DISCUSSION THREAD

**Discussion by danny1209** [OE4T/tegra-demo-distro#245] [Q&A]
Hi, 

I just compiled dunfell(L4T 32.7.3), the board can be rebooted successfully and I can see the GUI if it´s connected with monitor.
But no USB can be detected as usual
![image](https://user-images.githubusercontent.com/46848895/216164994-92c0428c-9a8d-4b86-aa04-5d35f25289d9.png)

that means my Jetson Nano after reboot is isolated, not possible to communicate with the computer per ssh.

**Comment by ichergui** [comment]
Hey @danny1209 
I don't understand your problem.
Do you mean that you are not able to access to the `serial UART` ? or you are not able to connect to your device via `SSH` ? 
I'm seeing that you are using a virtual machine. Check the virtual machine settings.
BTW, I don't recommend to use a virtual machine to avoid this kind of issues.

**Reply by danny1209** [reply]
Hey @ichergui 

thanks for your reply.
yes, it´s not able to access the serial UART.
VM works always well before. I did it before with L4T 32.7.2. Now I see it´s updated to 32.7.3. I´m not sure if it´s related.

**Reply by madisongh** [reply]
@danny1209 Sounds like you're expecting it to export USB gadgets after booting?   If that's the case, double-check that you included `l4t-usb-device-mode` in your build, and if there, check the system logs on the Nano to see if the udev rules are getting invoked.

**Reply by danny1209** [reply]
Hi @madisongh 
yes, it looks like this on my jetson nano
![image](https://user-images.githubusercontent.com/46848895/216327443-a2e03ea4-ab45-4025-a686-0138ec01ea78.png)

**Reply by madisongh** [reply]
Those are the USB host controller rules, not gadget rules.  The `l4t-usb-device-mode` recipe installs a file called `/etc/udev/rules.d/98-usb-gadget-tty.rules` and  a systemd service unit at `/lib/systemd/system/usbgx.service`.

**Reply by danny1209** [reply]
hi @madisongh 
thanks a lot! I added the recipe l4t-usb-device-mode and this solved my problem.
